### PR TITLE
WIP: Add play/pause support

### PR DESCRIPTION
**Pull request by vihu**
Hello @AndreasBackx 

I have been using this crate with bspwm/polybar with good results but was missing play/pause functionality (on click-left with polybar). I tried to work on it here a bit but ran into a couple roadblocks.

I moved some code around, specifically the spotify initialization itself into the Service module (so I could use it for LikedTracker as well as the new PlayPauseTracker).

I think it's mostly there but I could use some pointers in finishing this up. Please let me know how this draft looks so far.

Cheers!

### CODE REVIEW

File: src/client/service.rs [hunk 1]
```diff
@@ -24,4 +25,19 @@ impl Service {
...
+        println!("MARKER 2");
+        let mut client =
+            SpotifatiusClient::connect(format!("http://{ADDRESS}")).await
+            .context("Could not connect to monitor instance, make sure there is one running")?;
+        println!("MARKER 3 {:?}", client);
+        let request = tonic::Request::new(TogglePlayRequest {});
+        println!("MARKER 4 {:?}", request);
+        match client.toggle_play(request).await {
+            Ok(resp) => println!("MARKER 5 {:?}", resp),
+            Err(e) => println!("e: {:?}", e),
```
Comment (vihu) [inline src/client/service.rs:39]: We always end up here, why!?

File: src/server/grpc/server.rs [hunk 2]
```diff
@@ -75,4 +80,19 @@ impl Spotifatius for MySpotifatius {
             .map_err(|err| Status::internal(err.to_string()))?;
         Ok(Response::new(ToggleLikedResponse { is_liked }))
     }
+
+    async fn toggle_play(
+        &self,
+        _request: Request<TogglePlayRequest>,
+    ) -> Result<Response<TogglePlayResponse>, Status> {
+        println!("MARKER 6");
```
Comment (vihu) [inline src/server/grpc/server.rs:88]: We never get here!

File: src/client/service.rs (hunk 1 above)
Comment (AndreasBackx) [reply src/client/service.rs:39]: I don't know, the error will tell you what is wrong.

**Comment by AndreasBackx** [comment]
Thanks for your PR! It's a bit hard to review right now when the functionality is not really working. I don't currently have the time to spend to get a pull request working, though am happy to give a review once there is something working and help improve it.

It is however more important to talk about what this adds to the Spotifatius. Spotify is listening on the dbus for media controls, and so are other mpris media players. I have these shortcuts configured myself: https://github.com/AndreasBackx/dotfiles/blob/master/.config/sway/base#L288C1-L292

Sure, there might be a benefit to having them more easily accessible, though I personally do not see a benefit in integrating something like that in Spotifatius unless there is a good case for it. If you can give a good use case, I'm definitely willing to consider it.

**Comment by vihu** [comment]
> though I personally do not see a benefit in integrating something like that in Spotifatius unless there is a good case for it.

I agree actually, I didn't realize I could use playerctl (or some other existing module) to hook onto and get the next/prev/play-pause functionality. I stumbled upon this crate while looking for ways to connect to local spotify instance in Rust primarily. Thank you for the feedback, I don't particularly have a good reason to add my proposed functionality, therefore I'm going to close this.

**Pull request closed without merging**